src/types.ts

```diff
@@ -288,3 +293,5 @@ export type Options = RunOptions & {
   peerDependencies?: Index<any>,
   rcConfigPath?: string,
 }
+
+export type PreInitOptions = Omit<Options, 'packageData'> & { packageData?: string | PackageFile }
```

Comment:
Can we make adjustments so that `Options` can be used everywhere? It's confusing to have multiple options types.

Reply:
The issue here was that:
* before `initOptions` `packageData` can be string or PackageFile
* after `initOptions` it's definitely a string
and adding `| PackageFile` to the definition on `Options` lead to some (impossible) type errors later down the line.

I can see if there's a better way of doing it: I could hard cast everywhere to string, but I thought this was cleaner (though I agree it's not great).

I could try having a base class that doesn't have `packageData` at all -- that could be used almost everywhere, I expect.

Reply:
The "hard cast" versions adds casts to the calls to `runLocal` in `src/index.ts`:

```
await runLocal(pkgOptions, pkgData as string, pkgFile)
```

Let me know if you'd prefer that? I tend to avoid the casts where I can in TypeScript because it avoids the type checking, which can be an indicator that something's not right.

Reply:
Excellent points! I see there isn't an easy answer.

We can rule out the cast, since that defeats the purpose of type checking, as you said.

What do you think about c44df47? It seems that when `run` accepts `Options` then all the issues go away.

Reply:
That works and simplifies the code, but no longer tests the line 
```
...options.packageData && typeof options.packageData !== 'string' ? { packageData: JSON.stringify(options.packageData, null, 2) } : null,
```
in `initOptions.ts` (that does the PackageFile -> string conversion).

Reply:
You could keep the test as it was, cast to `any`, and keep the line tested by the unit tests, but not by the types? As a sort-of deprecated feature?

Reply:
You're right, and it gives `run` the wrong argument type. I do want to support `packageData: string | PackageFile` for module users.

I made another attempt in 3fcecba. I shift some options-related logic to `initOptions` and do the `PackageFile` conversion there. Let me know what you think.

Reply:
Looks good to me! You might want to go back to `test/index.test.ts` and pass `packageData` as a `PackageFile` instead of a string.

I merged in `main` to get it building.